pkg/controller/resourcequota/resource_quota_controller.go

```
@@ -228,18 +231,19 @@ func (rq *ResourceQuotaController) worker(queue workqueue.RateLimitingInterface)
 // Run begins quota controller using the specified number of workers
 func (rq *ResourceQuotaController) Run(workers int, stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
-	go rq.rqController.Run(stopCh)
-	// the controllers that replenish other resources to respond rapidly to state changes
-	for _, replenishmentController := range rq.replenishmentControllers {
-		go replenishmentController.Run(stopCh)
+
+	glog.Infof("Starting resource quota controller")
+
+	if !cache.WaitForCacheSync(stopCh, rq.informerSyncedFuncs...) {
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for caches to sync"))
```

what happens if we timeout?  it looks like we fail silently...